ibm/resource_ibm_is_virtual_endpoint_gateway_ip.go

```diff
@@ -0,0 +1,213 @@
+			isVirtualEndpointGatewayIPAddress: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Endpoint gateway IP address",
+			},
+			isVirtualEndpointGatewayIPTarget: {
+				Type:        schema.TypeList,
+				Computed:    true,
+				Description: "Endpoint gateway detail",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						isVirtualEndpointGatewayIPTargetID: {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The IPs target id",
+						},
+						isVirtualEndpointGatewayIPTargetName: {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The IPs target name",
+						},
+						isVirtualEndpointGatewayIPTargetResourceType: {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Endpoint gateway resource type",
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceIBMisVirtualEndpointGatewayIPCreate(d *schema.ResourceData, meta interface{}) error {
+	sess, err := vpcClient(meta)
+	if err != nil {
+		return err
+	}
+	gatewayID := d.Get(isVirtualEndpointGatewayID).(string)
+	ipID := d.Get(isVirtualEndpointGatewayIPID).(string)
+	opt := sess.NewAddEndpointGatewayIPOptions(gatewayID, ipID)
+	_, response, err := sess.AddEndpointGatewayIP(opt)
+	if err != nil {
+		log.Printf("Add Endpoint Gateway failed: %v", response)
+		return err
+	}
+	d.SetId(fmt.Sprintf("%s/%s", gatewayID, ipID))
+	return resourceIBMisVirtualEndpointGatewayIPRead(d, meta)
+}
+
+func resourceIBMisVirtualEndpointGatewayIPRead(d *schema.ResourceData, meta interface{}) error {
+	sess, err := vpcClient(meta)
+	if err != nil {
+		return err
+	}
+	parts, err := idParts(d.Id())
+	if err != nil {
+		return err
+	}
+	gatewayID := parts[0]
+	ipID := parts[1]
+	opt := sess.NewGetEndpointGatewayIPOptions(gatewayID, ipID)
+	result, response, err := sess.GetEndpointGatewayIP(opt)
+	if err != nil {
+		log.Printf("Get Endpoint Gateway IP failed: %v", response)
+		return err
+	}
+	d.Set(isVirtualEndpointGatewayIPID, result.ID)
+	d.Set(isVirtualEndpointGatewayIPName, result.Name)
+	d.Set(isVirtualEndpointGatewayIPAddress, result.Address)
+	d.Set(isVirtualEndpointGatewayIPCreatedAt, result.CreatedAt)
+	d.Set(isVirtualEndpointGatewayIPResourceType, result.ResourceType)
+	d.Set(isVirtualEndpointGatewayIPAutoDelete, result.AutoDelete)
+	d.Set(isVirtualEndpointGatewayIPTarget,
+		flattenEndpointGatewayIPTarget(result.Target.(*vpcv1.ReservedIPTarget)))
+	return nil
+}
+
+func resourceIBMisVirtualEndpointGatewayIPDelete(d *schema.ResourceData, meta interface{}) error {
+	sess, err := vpcClient(meta)
+	if err != nil {
+		return err
+	}
+	parts, err := idParts(d.Id())
+	if err != nil {
+		return err
+	}
+	gatewayID := parts[0]
+	ipID := parts[1]
+	opt := sess.NewRemoveEndpointGatewayIPOptions(gatewayID, ipID)
+	response, err := sess.RemoveEndpointGatewayIP(opt)
+	if err != nil {
+		log.Printf("Remove Endpoint Gateway IP failed: %v", response)
```

Comment:
Check for 404 condition also and set d.SetID("")

Reply:
handled